testinfra/modules/x509.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+from datetime import datetime
+from testinfra.modules.base import Module
+from testinfra.utils import cached_property
+
+
+class X509(Module):
+
+    def __init__(self, name):
+        self.name = name
+        super(X509, self).__init__()
+
+    @property
+    def issuer(self):
+        """Return issuer of certificate
+
+        >>> host.x509('/etc/pki/tls/certs/dummy.crt').issuer
+        """
+
+        raise NotImplementedError
+
+    @property
+    def email(self):
+        """Return email of certificate
+
+        >>> host.x509('/etc/pki/tls/certs/dummy.crt').email
+        """
+
+        raise NotImplementedError
+
+    @property
+    def subject(self):
+        """Return subject of certificate
+
+        >>> host.x509('/etc/pki/tls/certs/dummy.crt').subject
+        """
+
+        raise NotImplementedError
+
+    @property
+    def enddate(self):
+        """Return enddate as datetime of certificate
+
+        >>> host.x509('/etc/pki/tls/certs/dummy.crt').enddate
+        """
+
+        raise NotImplementedError
+
+    @property
+    def serial(self):
+        """Return serial of certificate
+
+        >>> host.x509('/etc/pki/tls/certs/dummy.crt').serial
+        """
+
+        raise NotImplementedError
+
+    def __repr__(self):
+        return "<x509 %s>" % (self.name,)
+
+    @classmethod
+    def get_module_class(cls, host):
+        return OpenSSL
+
+
+class OpenSSL(X509):
+
+    @cached_property
+    def _openssl(self):
+        return self.find_command('openssl')
+
+    @property
+    def issuer(self):
+        stdout = self.check_output(
+            '%s x509 -in %s -noout -issuer', self._openssl, self.name
+        )
+        return stdout
```

Comment:
`issuer` and `subject` doesn't work for me:

```
import testinfra
host = testinfra.get_host('local://')
c = host.x509('/etc/ssl/certs/ssl-cert-snakeoil.pem')
for a in ('issuer', 'email', 'subject', 'enddate', 'serial'):
    print(a, repr(getattr(c, a)))
```

result in

```
issuer 'issuer=CN = aldo.in.philpep.org'
email ''
subject 'subject=CN = aldo.in.philpep.org'
enddate datetime.datetime(2023, 10, 31, 22, 51, 18)
serial 'A18E6C82C49B31EA'
```

I'm using openssl 1.1.1c-1 on debian buster.